reckless_github/src/repository.rs

```diff
@@ -35,6 +37,10 @@ impl Repository for Github {
     /// Where to store the index is an implementation
     /// details.
     async fn init(&self) -> Result<(), RecklessError> {
+        let repo = match git2::Repository::clone(&self.url, get_dir_path_from_url(&self.url)) {
+            Ok(repo) => repo,
+            Err(e) => panic!("failed to open: {}", e),
```

Comment:
```suggestion
            Err(e) => return RecklessError::new(format!("failed to open: {}"));,
```

Or something like that I do not know if this compile, do not panic, but return an error instead